### PR TITLE
Anti url flood

### PR DESCRIPTION
**Pull request by m-242**

### CODE REVIEW

File: bitbot/links_test.go [hunk 1]
```diff
@@ -60,3 +60,19 @@ func TestGetHTMLTitleWithEmptyTitle(t *testing.T) {
 		t.Fail()
 	}
 }
+
+func TestAntiUrlFlood(t *testing.T) {
+	t.Log("Note : If the previous test failed, this one will hang indefinitely")
+
+	message := "https://secops.space/"
```
Comment (bbriggs) [inline bitbot/links_test.go]: Test shouldn't rely on external infrastructure to work.  Might want to mock a small http server here and call a lookup on localhost.

File: bitbot/links.go [hunk 2]
```diff
@@ -17,25 +18,31 @@ var URLReaderTrigger = NamedTrigger{
...
 		resp := lookupPageTitle(m.Content)
 		if resp != "" {
-			irc.Reply(m, lookupPageTitle(m.Content))
+			irc.Reply(m, resp)
 		}
 		return true
 	},
 }
 
+var lastTimeTitleLookup = time.Date(
```
Comment (bbriggs) [inline bitbot/links.go]: Setting this var globally is a bad idea. Anything could access and modify it.

File: bitbot/links.go [hunk 3]
```diff
@@ -17,25 +18,31 @@ var URLReaderTrigger = NamedTrigger{
...
+        1970, 1, 1, 0, 0, 0, 0, time.UTC)
+
 func isURL(message string) bool {
 	return xurls.Strict().MatchString(message)
 }
 
 func lookupPageTitle(message string) string {
+	if time.Now().Sub(lastTimeTitleLookup).Minutes() < 2 {
```
Comment (bbriggs) [inline bitbot/links.go]: This function seems to put a space of two minutes between _any two URL lookups_, which is not what we were shooting for. We were trying to cache the URL title if it had been recently looked up and return that back to chat without performing unnecessary queries.

**Comment by m-242** [comment]
This pull request was written while fixing the wrong issue, it's better to start with a clean slate IMO.

**Pull request closed without merging**